runtime/cudaq/distributed/builtin/activate_custom_mpi.sh

```diff
@@ -0,0 +1,44 @@
+#!/bin/bash
+
+# ============================================================================ #
+# Copyright (c) 2022 - 2023 NVIDIA Corporation & Affiliates.                   #
+# All rights reserved.                                                         #
+#                                                                              #
+# This source code and the accompanying materials are made available under     #
+# the terms of the Apache License 2.0 which accompanies this distribution.     #
+# ============================================================================ #
+
+# This script will build and activate a **custom** CUDA Quantum MPI interface.
+# Specifically, this script builds a MPI plugin interface against your local MPI installation 
+# and exports an environment variable $CUDAQ_MPI_COMM_LIB to tell CUDA Quantum to use this MPI Communicator plugin. 
+#
+# It requires a GNU C++ compiler (g++).
+#
+# Please check/set the following environment variables:
+#  - $MPI_PATH = Path to your MPI library installation directory.
+#                If your MPI library is installed in system
+#                directories as opposed to its own (root) directory,
+#                ${MPI_PATH}/include is expected to contain the mpi.h header.
+#                ${MPI_PATH}/lib64 or ${MPI_PATH}/lib is expected to contain libmpi.so.
+#
+# Run (inside this directory): source ./activate_custom_mpi.sh
+# 
+# You could add $CUDAQ_MPI_COMM_LIB to your ~/.bashrc file to persist the environment variable.
```

Comment:
Hm, does that mean if they persist it, a user would have to run this script ever time the use CUDA Quantum in a new shell? We should make sure setting CUDAQ_MPI_COMM_LIB manually is not required once the plugin is built.

Reply:
One option is that in the logic defined [here](https://github.com/NVIDIA/cuda-quantum/blob/942267d549dd1fc4b4ae6eea024191d42f8b33cb/runtime/cudaq/cudaq.cpp#L32C5-L84) we add `<install_dir>/distributed_interface/<comm_lib.so>` to the search list (after environment variable but before any others).
i.e., if the user has activated it once (and left the `.so` file in the default location), it will be always used the next time. 

What do you think?

Reply:
Yes, I think that would be good.

Reply:
I've added `<install_dir>/distributed_interfaces` to the plugin search path in  https://github.com/NVIDIA/cuda-quantum/pull/966/commits/427a63ce4415933756cf6f33c996a770e5384825